src/defmt.rs

```diff
@@ -244,6 +246,15 @@ pub fn decode_rtt<W: Write>(
         }
 
         for (secondary_index, core) in secondary_cores.iter_mut().enumerate() {
+            log::debug!("Secondary core index: {}, active cores: {}", secondary_index, active_cores);
+            // eww
+            let active_secondary_cores = (active_cores - 1).into();
+            let secondary_core_id = secondary_index + 1;
+            if secondary_core_id > active_secondary_cores {
+                // Core is not active so ignore it and all the next ones
+                log::debug!("Breaking away");
+                break;
+            }
```

Comment:
Please replace this with [`Iterator::take()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take).

Reply:
I have implemented the change. I still feel that, considering the code examples available, it is more desirable to have only one core by default, but I guess for backward compatibility it is better to leave things like that.